modules/storages/app/contracts/storages/storages/one_drive_contract.rb

```diff
@@ -32,5 +32,9 @@ module Storages::Storages
   class OneDriveContract < ::ModelContract
     attribute :host
     validates :host, absence: true
+    attribute :tenant_id
+    validates :tenant_id, format: { with: /\A(?:[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}|consumers)\z/i }
+    attribute :drive_id
+    validates :drive_id, format: { with: /\A[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}\z/i }, allow_nil: true
```

Comment:
🔴 why is the drive id allowed to be `nil`? at least some of our queries will fail on a `nil` value here.

Reply:
This is due to the fact, how the solution is build. The contract is evaluated on both `create` and `update`. Therefor and because the drive ID is inserted via the `update` action on the second `page`, the `create` action is called with an empty drive ID. That is of cause not ideal, but it will also be reworked with the refactoring of the UI to Primer.

Reply:
Okay, that is a bit cracked, but I'd accept it due to the intermediate state this implementation has (it is to be replaced by primer form very soon).